test/jdk/jdk/jfr/event/io/TestSerializationMisdeclarationEvent.java

```diff
@@ -0,0 +1,259 @@
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+/*
+ * @test
+ * @bug 8275338
+ * @summary Check generation of JFR events for misdeclared fields and methods
+ *          relevant to serialization
+ * @modules java.base/jdk.internal.event
+ * @key jfr
+ * @requires vm.hasJFR
+ * @library /test/lib
+ * @run junit/othervm jdk.jfr.event.io.TestSerializationMisdeclarationEvent
+ */
+public class TestSerializationMisdeclarationEvent {
+
+    private static final List<RecordedEvent> events;
+
+    static Arguments[] testSingleClassMisdeclarations() {
+        return new Arguments[] {
+                arguments(NoSUID.class, SUID_EXPLICIT),
+                arguments(NoSUID.class, SER_PERS_NOT_NULL),
+
+                arguments(NonLongSUID.class, SUID_LONG),
+                arguments(NonLongSUID.class, SUID_CONVERTIBLE_TO_LONG),
+
+                arguments(EnumClass.class, SUID_INEFFECTIVE_ENUM),
+                arguments(EnumClass.class, SUID_PRIVATE),
+                arguments(EnumClass.class, SUID_LONG),
+                arguments(EnumClass.class, SUID_STATIC),
+                arguments(EnumClass.class, SUID_FINAL),
+                arguments(EnumClass.class, SER_PERS_INEFFECTIVE_ENUM),
+                arguments(EnumClass.class, SER_PERS_PRIVATE),
+                arguments(EnumClass.class, SER_PERS_STATIC),
+                arguments(EnumClass.class, SER_PERS_FINAL),
+                arguments(EnumClass.class, SER_PERS_TYPE_OSF_ARRAY),
+                arguments(EnumClass.class, PRIV_METH_PRIV),
+                arguments(EnumClass.class, PRIV_METH_NON_STATIC),
+                arguments(EnumClass.class, PRIV_METH_RET_TYPE),
+                arguments(EnumClass.class, PRIV_METH_PARAM_TYPES),
+                arguments(EnumClass.class, PRIV_METH_INEFFECTIVE_ENUM),
+                arguments(EnumClass.class, ACC_METH_INEFFECTIVE_ENUM),
+
+                arguments(RecordClass.class, SER_PERS_INEFFECTIVE_RECORD),
+                arguments(RecordClass.class, SER_PERS_TYPE_OSF_ARRAY),
+                arguments(RecordClass.class, SER_PERS_VALUE_OSF_ARRAY),
+                arguments(RecordClass.class, PRIV_METH_INEFFECTIVE_RECORD),
+
+                arguments(C.class, ACC_METH_NON_ACCESSIBLE),
+
+                arguments(Acc.class, ACC_METH_NON_ABSTRACT),
+                arguments(Acc.class, ACC_METH_NON_STATIC),
+                arguments(Acc.class, ACC_METH_RET_TYPE),
+                arguments(Acc.class, ACC_METH_PARAM_TYPES),
+        };
+    }
+
+    static Arguments[] testGoodClass() {
+        return new Arguments[] {
+                arguments(A.class),
+                arguments(B.class),
+        };
+    }
+
+
+    @ParameterizedTest
+    @MethodSource
+    public void testSingleClassMisdeclarations(Class<?> cls, int kind) {
+        singleClassEvent(cls, kind);
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void testGoodClass(Class<?> cls) {
+        assertEquals(0, getEventsFor(cls).size());
```

Comment:
Can this and other assertions in this test be updated to include the class name which failed? You can still use `assertEquals(...)`, it takes an optional message as a third parameter which you could use to include the failing class name. It becomes a bit more easier to debug (unexpected) failures when the assertion includes these additional details.

Reply:
Thanks for the suggestion, will take a look.